modules/infoicon/Infoicon.tsx

```diff
@@ -1,48 +1,77 @@
-import React from "react";
+import React, { PropsWithChildren } from "react";
 import { useRegistry } from "@diana-ui/hooks";
 import { withStyles } from "@diana-ui/base";
 import { ThemeStyleSheetFactory, WithStylesProps } from "@diana-ui/types";
-import { DescriptionMedium } from "@diana-ui/typography";
+import { DescriptionMedium, BodyHighlight } from "@diana-ui/typography";
+import { IAvatarProps } from "@diana-ui/avatar";
 
-export interface IProps {
-  title: string;
+export interface IInfoIconProps {
+  title?: string;
+  children?: string | JSX.Element;
   withPadding?: boolean;
+  vertical?: boolean;
+  className?: string;
+  avatarComponentName?: string;
+  avatarOptions?: any;
 }
 
 const styleSheet: ThemeStyleSheetFactory = theme => ({
   infoicon: {
-    height: 80,
-    display: "flex"
-  },
-  wrapper: {
+    height: "auto",
     display: "flex"
   },
+  vertical: {},
   withpadding: {
     border: "1px solid lightgrey",
     borderRadius: 10,
     padding: "14px 24px"
   },
   text: {
     display: "flex",
-    marginLeft: 8
-  }
+    marginLeft: theme.spaceUnit.md,
+    flexDirection: "column",
+    paddingTop: theme.spaceUnit.xxs
+  },
+  textVertical: {},
+  title: {},
+  body: {}
 });
 
-const Infoicon: React.FC<IProps & WithStylesProps> = ({
+type IProps = IInfoIconProps & WithStylesProps;
+
+const Infoicon: React.FC<IProps> = ({
   cx,
   styles,
   title,
-  withPadding = false
+  children,
+  withPadding = false,
+  vertical = false,
+  className,
+  avatarComponentName = "Avatar",
+  avatarOptions
 }) => {
-  const StyledAvatar = useRegistry("Avatar");
+  const StyledAvatar: any = useRegistry(avatarComponentName);
 
   return (
-    <div className={cx(styles.infoicon)}>
-      <div className={cx(styles.wrapper, withPadding && styles.withpadding)}>
-        <StyledAvatar />
-        <div className={cx(styles.text)}>
-          {title && <DescriptionMedium>{title}</DescriptionMedium>}
-        </div>
+    <div
+      className={cx(
+        styles.infoicon,
+        withPadding && styles.withpadding,
+        vertical && styles.vertical,
```

Comment:
Instead of having `vertical` and `textVertical`, do you think it would be better to have a modifier class (e.g. `.vertical`)? We could then add the necessary modified/ extra styles for each element that needs it with `"@selectors": { "&.vertical": { ...styles } }`